Frontend/CO.CDP.OrganisationApp/Constants/ApiExceptionMapper.cs

```diff
@@ -42,6 +44,7 @@ private static string GetErrorMessageByCode(string errorCode)
             ErrorCodes.UNKNOWN_ORGANISATION => ErrorMessagesList.UnknownOrganisation,
             ErrorCodes.BUYER_INFO_NOT_EXISTS => ErrorMessagesList.BuyerInfoNotExists,
             ErrorCodes.UNKNOWN_BUYER_INFORMATION_UPDATE_TYPE => ErrorMessagesList.UnknownBuyerInformationUpdateType,
+            ErrorCodes.ORGANISATION_UPDATE_INVALID_INPUT => exceptionMessage ?? ErrorMessagesList.UnexpectedError,
```

Comment:
Do I understand this correctly, that this way we'll expose the message returned by the API?

If that's the case, we initially disregarded this approach and decided APIs should return error codes that are mapped to a friendly message by the frontend. What has changed?

Reply:
My concern is about using the exception message in the frontend at all. We are not supposed to do that according to the approach with error codes we agreed to.

So let me repeat the question: Do I understand this correctly, that this way we'll expose the message returned by the API?

Using null forgiving is definitely not the right solution here. Do we even expose the `Detail` in `Production`?

Reply:
message coming from the API will be exposed to frontend especially for "InvalidUpdateOrganisationCommand" exception class. As this class is used to throw multiple messages based on the validation, mostly Invalid input.  
It is very difficult to map one exception with multiple message at front end.  and I couldn't find a better way to show user friendly message unless if we just want to show "Invalid Input".

Reply:
We agreed we should not expose the api messages. Error codes should be mapped to messages by the frontend. The Detail you reference is not even populated in Production. You need a more specific error code returned by the API.

Reply:
It will be at least 10 new exception classes added, let me ask business if they are happy with generic message "Invalid Input"

Reply:
This has much bigger implications. Messages need to be translated. We do not translate API messages, and it would have a huge impact if we wanted to start.

Reply:
Not even mentioning the API isn't helpful at all by returning a generic error code in all validation cases.

By the way, you do not need a new exception class for each code (although it has its benefits).

If you wanted to avoid it, you could provide factories for exceptions:

```csharp
 throw InvalidUpdateOrganisationCommand.MissingIdentifierNumberForScheme(identifier.Scheme);
```

The exception would be created with code:

```csharp
public class InvalidUpdateOrganisationCommand(string message, string code = "INVALID_UPDATE_ORGANISATION_COMMAND", Exception? cause = null) : Exception(message, cause)
{
    public static InvalidUpdateOrganisationCommand MissingIdentifierNumberForScheme(string scheme)
    {
        return new InvalidUpdateOrganisationCommand($"Missing Identifier Number for scheme '{scheme}'", "MISSING_IDENTIFIER_NUMBER");
    }
}
```

The code then can be returned from the API. 

I think it's important that the code is encapsulated rather than passed from the outside.


However, not sure if it's much better than:

```csharp
 throw new InvalidUpdateOrganisationCommand.MissingIdentifierNumberForScheme(identifier.Scheme);
```

```csharp
public class InvalidUpdateOrganisationCommand(string message, Exception? cause = null) : Exception(message, cause)
{
    public class MissingIdentifierNumberForScheme(string scheme)
        : InvalidUpdateOrganisationCommand($"Missing Identifier Number for scheme '{scheme}'.");
}
```